constraints/match.js

```diff
@@ -24,26 +24,52 @@ function getLeaves(x)
 	return leaves;
 }
 
-function sameIds(a1, a2)
+function sameIdsOrdered(a1, a2)
 //helper function to compare two arrays of children
 //since there isn't a built_in array comparator.
 {
 	if(a1.length !== a2.length)
 		return false;
-	
+
 	var i = 0;
 	while(i<a1.length){
 		if(a1[i].id !== a2[i].id)
 			return false;
 		i++;
 	}
-	
+
+	return true;
+}
+
+/* function to compare sets of terminals {A} and {B}. returns true iff for each
+ * element in A, there is an element in B with the same value for the property
+ * "id" and A and B are of the same lenght.
+ * Order insensitive version of sameIdsOrdered.
+ */
+function sameIds(a1, a2){
+	if (a1.length !== a2.length){
+		return false;
+	}
+	for (var x = 0; x < a1.length; x ++){ // for each element in a1
+		var y = 0;
+		var matched = false; // keeps track of if a1[x] has a match in a2
+		while (matched == false){//there is an element in a2 ...
```

Comment:
I'm going to give in to my impulse to nitpick here and suggest:
while(matched == false && y < a2.length)